test/Microsoft.TestPlatform.Common.UnitTests/Filtering/FastFilterTests.cs

```diff
@@ -90,6 +90,20 @@ public void FastFilterWithSingleEqualsClause()
         Assert.IsFalse(fastFilter.Evaluate(s => "Test2"));
     }
 
+    [TestMethod]
+    public void AvoidingStackOverFlow()
+    {
+        var testCaseFilter = "Test1";
```

Comment:
You are not reusing this variable so you could directly hardcode the value inside `FilterExpressionWrapper` ctor.